.github/workflows/build_test_ci.yml

```diff
@@ -89,6 +89,59 @@ jobs:
     - name: Nilcheck
       run: make nilcheck
 
+  e2e-test:
+    needs:  [go-build-test, docker-build]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Harden Runner
+      uses: step-security/harden-runner@v2
+      with:
+        disable-sudo: true
+        egress-policy: block
+        allowed-endpoints: >
+          api.linode.com:443
+          api.github.com:443
+          github.com:443
+          gcr.io:443
+          proxy.golang.org:443
+          sum.golang.org:443
+          *.githubusercontent.com:443
+          docker.io:443
+          registry-1.docker.io:443
+          auth.docker.io:443
+          production.cloudflare.docker.com:443
+          storage.googleapis.com:443
+          registry.k8s.io:443
+          *.pkg.dev:443
+          *.amazonaws.com:443
+          *.blob.core.windows.net:443
+          quay.io:443
+          *.quay.io:443
+          api.snapcraft.io:443
+
+    - uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: 'stable'
+
+    - name: Docker cache
+      uses: ScribeMD/docker-cache@0.3.7
+      with:
+        key: docker-${{ runner.os }}-${{ hashFiles('go.sum') }}}
+
+    - name: E2E test
+      run: make e2etest
+      env:
+        LINODE_TOKEN: ${{ secrets.LINODE_TOKEN }}
```

Comment:
this has been created now and you can test, let me know if you run into any problems with the token i can get them resolved for you